programs/svm-spoke/src/instructions/deposit.rs

```diff
@@ -101,21 +102,39 @@ pub fn _deposit(
         if exclusivity_deadline <= MAX_EXCLUSIVITY_PERIOD_SECONDS {
             exclusivity_deadline += current_time;
         }
-
         if exclusive_relayer == Pubkey::default() {
             return err!(CommonError::InvalidExclusiveRelayer);
         }
     }
 
-    // Depositor must have delegated input_amount to the state PDA.
+    // Verify delegate PDA
+    let seed_hash = derive_deposit_delegate_seed_hash(
```

Comment:
As we don't include `quote_timestamp` and `fill_deadline` this would still leave room for some griefing attacks where a deposit could become practically unfillable if someone changed those parameters using a delegation from a separate transaction.

Since the above params could be hard to predict in `deposit_now` one alternative would be to calculate `seed_hash` at the entry of the handler for each of the deposit flavors and pass it as parameter to the common implementation.

Reply:
Yeah, we’d still have to calculate it off-chain and pass in the delegate PDA, which depends on current_time in deposit_now, and that’s the tricky part, right? So I don’t think that really solves the problem.